content/general/concepts/abstract-factory/abstract-factory.md

```diff
@@ -0,0 +1,381 @@
+    }
+  }
+  // Enforces concrete classes to override this method
+  public abstract void increaseOverdraft(int requestedLimit);
+
+  // Getters
+}
+```
+
+The parent class above, as well as providing the common fields and constructor, requires its children to implement the `increaseOverdraft` method. This is the differentiating feature described between each current account.
+
+Below provides the concrete implementations of `CurrentAccount` for this example:
+
+```java
+public class GoldPersonal extends CurrentAccount {
+  // The Gold Personal has a high max overdraft limit
+  public static final int MAX_OVERDRAFT = 3500;
+
+  public GoldPersonal(Customer accountHolder, int initBalance, int agreedOverdraftLimit) {
+    super(accountHolder, initBalance, agreedOverdraftLimit);
+  }
+  // Concrete overridden method
+  @Override
+  public void increaseOverdraft(int requestedLimit) {
+    if (requestedLimit < MAX_OVERDRAFT) {
+      this.setAgreedOverdraftLimit(requestedLimit);
+    }
+  }
+}
+```
+
+```java
+public class SilverPersonal extends CurrentAccount {
+  // The Silver Personal has a mid max overdraft limit
+  public static final int MAX_OVERDRAFT = 1200;
+
+  public SilverPersonal(Customer accountHolder, int initBalance, int agreedOverdraftLimit) {
+    super(accountHolder, initBalance, agreedOverdraftLimit);
+  }
+  // Concrete overridden method
+  @Override
+  public void increaseOverdraft(int requestedLimit) {
+    if (requestedLimit < MAX_OVERDRAFT) {
+      this.setAgreedOverdraftLimit(requestedLimit);
+    }
+  }
+}
+```
+
+```java
+public class BronzePersonal extends CurrentAccount {
+  // The Bronze Personal has a low max overdraft limit
+  public static final int MAX_OVERDRAFT = 500;
+
+  public BronzePersonal(Customer accountHolder, int initBalance, int agreedOverdraftLimit) {
+    super(accountHolder, initBalance, agreedOverdraftLimit);
+  }
+  // Concrete overridden method
+  @Override
+  public void increaseOverdraft(int requestedLimit) {
+    if (requestedLimit < MAX_OVERDRAFT) {
+      this.setAgreedOverdraftLimit(requestedLimit);
+    }
+  }
+}
+```
+
+```java
+public class CreditBuilder extends CurrentAccount {
+  // The Credit Builder does not allow for an overdraft
+  public static final int MAX_OVERDRAFT = 0;
+
+  public CreditBuilder(Customer accountHolder, int initBalance, int agreedOverdraftLimit) {
+    super(accountHolder, initBalance, agreedOverdraftLimit);
+  }
+  // Concrete overridden method
+  @Override
+  public void increaseOverdraft(int requestedLimit) {
+    if (requestedLimit < MAX_OVERDRAFT) {
+      this.setAgreedOverdraftLimit(requestedLimit);
+    }
+  }
+}
+```
+
+In each concrete current account, a `static` variable (`MAX_OVERDRAFT`) hase been provided and constructed with the correct value. This is then used in the `increaseOverdraft` method to ensure this limit is not exceeded.
+
+Below provides an abstract model `SavingsAccount` class for its concrete sub-classes to be based on:
```

Comment:
Here, you wrote `SavingsAccount`, but most, if not all, of the mentions after this are `SavingAccount`. 

Should we change this to `SavingAccount` or the other way around?